syntax = "proto3";

package browser_copilot;

// Message types that can be sent to the service
enum MessageType {
  TEXT = 0;
  IMAGE = 1;
  PDF = 2;
}

// Request to send a message to the agent
message SendMessageRequest {
  string session_id = 1;      // Session identifier for conversation continuity
  MessageType message_type = 2; // Type of message being sent
  string content = 3;         // Text content (for TEXT messages) or file path (for IMAGE/PDF)
}

// Response streamed back from the service
message MessageResponse {
  oneof response {
    string text = 1;          // Text message response
    ImageResponse image = 2;  // Image response
  }
}

// Image response with file path
message ImageResponse {
  string file_path = 1;      // Path to the image file
  string description = 2;    // Optional description of the image
}

// Service definition
service BrowserCopilotService {
  // Server-side streaming RPC to send messages and receive responses
  rpc SendMessage(SendMessageRequest) returns (stream MessageResponse);
}

